docs/src/main/asciidoc/qute.adoc

```diff
@@ -0,0 +1,271 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Qute Templating Engine
+
+include::./attributes.adoc[]
+
+Qute is a templating engine designed specifically to meet the Quarkus needs.
+The usage of reflection is minimized to reduce the size of native images.
+The API combines both the imperative and the non-blocking reactive style of coding.
+In the development mode, all files located in `META-INF/resources/templates` are watched for changes and modifications are immediately visible.
+Furthermore, we try to detect most of the template problems at build time.
+In this guide, you will learn how to easily render templates in your application.
+
+[NOTE]
+====
+This extension is considered `preview`.
+API or configuration properties might change as the extension matures.
+Feedback is welcome on our https://groups.google.com/d/forum/quarkus-dev[mailing list] or as issues in our https://github.com/quarkusio/quarkus/issues[GitHub issue tracker].
+====
+
+== Hello World with JAX-RS
+
+If you want to use Qute in your JAX-RS application, you need to add the `quarkus-qute-resteasy` extension first.
+In your `pom.xml` file, add:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-qute-resteasy</artifactId>
+</dependency>
+----
+
+We'll start with a very simple template:
+
+.hello.txt
+----
+Hello {name}! <1>
+----
+<1> `{name}` is a value expression that is evaluated when the template is rendered.
+
+NOTE: By default, all files located in the `META-INF/resources/templates` directory and its subdirectories are registered as templates. Templates are validated during startup and watched for changes in the development mode.
+
+Now let's inject the "compiled" template in the resource class.
+
+.HelloResource.java
+[source,java]
+----
+package org.acme.quarkus.sample;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+import io.quarkus.qute.Rendering;
+import io.quarkus.qute.Template;
+
+@Path("hello")
+public class HelloResource {
+
+    @Inject
+    Template hello; <1>
```

Comment:
Didn't we have a variant where I didn't need to inject anything and it would default to the "class/method" template name?

Reply:
Nope. It was part of your proposal but I was against it ;-).

Reply:
Well, I'm still in favour of it, so can we have it or do I need to send a PR? Of course we can keep this variant, but I don't feel this scales if you have more than one method doing rendering.

Reply:
I think that the right approach would be to 1) create an issue and 2) send a PR after this one gets merged. Thanks for understanding ;-)

Reply:
Absolutely, which is why none of the issues I raised here should block this PR.